skopt/space.py

```diff
@@ -324,6 +352,13 @@ def __init__(self, dimensions):
 
         self.dimensions = _dimensions
 
+    def __eq__(self, other):
+        return all([a == b for a, b in zip(self.dimensions, other.dimensions)])
+
+    def __repr__(self):
+        return "Space([{}])".format(
```

Comment:
limiting the number of dimensions I'm not so sure about, even if you have a space with lots of dimensions you probably still want to see all, no?